src/components/form-elements/draft-js-mention-selector/createMentionSelectorState.js

```diff
@@ -0,0 +1,45 @@
+// @flow
+import { ContentState, EditorState, Modifier, SelectionState } from 'draft-js';
+import DraftMentionDecorator from './DraftMentionDecorator';
+
+// returns data for first mention in a string
+const getMentionFromText = (text: string) => {
+    // RegEx.exec() is stateful, so we create a new regex instance each time
+    const mentionRegex = /([@＠﹫])\[([0-9]+):([^\]]+)]/gi;
+    const matchArray = mentionRegex.exec(text);
+    if (!matchArray) {
+        return null;
+    }
+    const [fullMatch, triggerSymbol, id, name] = matchArray;
+    const start = matchArray.index;
+    const end = start + fullMatch.length;
+    const data = { id, name, content: `${triggerSymbol}${name}` };
+    return { start, end, data };
+};
+
+// creates draftjs state with mentions parsed into entities
+const createMentionSelectorState = (message: string = '') => {
+    let contentState = ContentState.createFromText(message);
+    let contentBlock = contentState.getFirstBlock();
+
+    while (contentBlock != null) {
+        const text = contentBlock.getText();
+        const mention = text ? getMentionFromText(text) : null;
+        if (mention == null) {
```

Comment:
why truthy/falsy? what kinds of things can `mention` and `contentBlock` be?

Reply:
`== null` is only undefined/null, not falsy